### PR TITLE
better psr4

### PR DESCRIPTION
**Pull request by TomasVotruba**
- Move findNonAnonymousClassLikes to BetterNodeFinder
- NormalizeNamespaceByPSR4ComposerAutoloadRector: Replace references correctly
- rebase
- fixup! rebase
- fixup! fixup! rebase

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -30,6 +30,9 @@ jobs:
                     - 'e2e/parallel-custom-config'
                     - 'e2e/parallel-reflection-resolver'
                     - 'e2e/plain-views'
+                    - 'e2e/psr-4'
+                    - 'e2e/template-extends'
+                    - 'e2e/use-rector-configurator'
```
Comment (samsonasik) [inline .github/workflows/e2e.yaml]: `e2e/template-extends` and `e2e/use-rector-configurator` no longer exists, can be removed

**Comment by TomasVotruba** [comment]
Reproducer script:

```bash
cd e2e/psr-4 && php ../e2eTestRunner.php
```

**Comment by samsonasik** [comment]
composer install need to be run inside:

```
cd e2e/psr-4
composer install
php ../e2eTestRunner.php

**Comment by TomasVotruba** [comment]
@samsonasik Indeed :+1: 

I'm looking into it and the class rename should happen. The array is always set:

```bash
array:2 [
  "Test\BarClass" => "Test\Bar\BarClass"
  "Test\FooClass" => "Test\Foo\FooClass"
]
```

 I wonder what is wrong there :thinking:

**Comment by TomasVotruba** [comment]
I see. First the `Tests\FooClass` is renamed to `Tests\Foo\FooClass`, so these references are updated later by PostRector that runs after Rector run.
That means only 1 class name is renamed.

Only then the `Tests\BarClass` is renamed, and then again post rector is run again, with both class renames.

That means the `ClassRenamingPostRector` always renames the current file, with rename so far. 


---

That is our architecture at the moment. The post rectors are not run after all the files and rules, but once in the end of every file. We'd have to change it to run this single post rector after all the others files + rectors.

**Comment by TomasVotruba** [comment]
Closing as the buggy rule is now removed to avoid causing harm.

**Pull request closed without merging**